apps/api/src/services/agent-manager.service.ts

```diff
@@ -753,4 +758,38 @@ export class AgentManager {
       return { agents: [], total: 0 };
     }
   }
+
+  /**
+   * Get agents with paging and filtering
+   */
+  async getAgents({
+    filter,
+    pagingParams,
+  }: {
+    filter?: string;
+    pagingParams: PagingParams;
+  }) {
+    if (filter?.length === 42) {
+      return findByWallet({ walletAddress: filter, pagingParams });
+    }
+    if (typeof filter === "string" && filter.length > 0) {
+      return findByName({ name: filter, pagingParams });
+    }
+
+    return findAll(pagingParams);
```

Comment:
do you know if `findAll` automatically filters out the agent `apiKey` field? i'm guessing it _doesn't_...so the API calls that use this have to filter out that value like [here](https://github.com/recallnet/js-recall/blob/33d3e5478e7202d61a4e466b08ce9770d3c24d53/apps/api/src/controllers/user.controller.ts#L202).

if this is true, it's obv not ideal for internal DX...we should address in a separate issue. most likely, we'll have a [separate table for API keys](https://www.notion.so/Comps-db-API-redesign-1fbdfc9427de809c8697ee92e74ac752?pvs=4#1fbdfc9427de8069b3d1c4760f4cd53c)

Reply:
cool, created this as a follow up https://github.com/recallnet/js-recall/issues/450